tsweb/tsweb.go

```diff
@@ -463,3 +505,42 @@ func writeMemstats(w io.Writer, ms *runtime.MemStats) {
 	c("frees", ms.Frees, "cumulative count of heap objects freed")
 	c("num_gc", uint64(ms.NumGC), "number of completed GC cycles")
 }
+
+func foreachExportedStructField(rv reflect.Value, f func(fieldOrJSONName, promTyp string, rv reflect.Value)) {
+	t := rv.Type()
+	for i, n := 0, t.NumField(); i < n; i++ {
+		sf := t.Field(i)
+		name := sf.Name
```

Comment:
Do we need any special handling of embedded fields here? I guess it's OK to have the embedded field be a part of the naming scheme.

Reply:
yup. fine by me.